test/test_iterator.cpp

```diff
@@ -237,8 +237,10 @@ namespace sparrow
 
         TEST_CASE_FIXTURE(iterator_fixture, "contiguous iterator")
         {
+            #ifndef EMSCRIPTEN
```

Comment:
Why is this bypass necessary? That code should pass on all platforms, assuming a standard library with concepts.

Reply:
it fails with the following compilation error
```
/Users/thorstenbeier/src/pyjs/emsdk_install/upstream/emscripten/cache/sysroot/include/c++/v1/__memory/pointer_traits.h:118:22: error: implicit instantiation of undefined template 'std::__pointer_traits_element_type<sparrow::test_iterator>'
  118 |     typedef typename __pointer_traits_element_type<pointer>::type    element_type;
      |                      ^
/Users/thorstenbeier/src/pyjs/emsdk_install/upstream/emscripten/cache/sysroot/include/c++/v1/__memory/pointer_traits.h:182:20: note: in instantiation of template class 'std::pointer_traits<sparrow::test_iterator>' requested here
  182 |     decltype((void)pointer_traits<_Pointer>::to_address(std::declval<const _Pointer&>()))
      |                    ^
/Users/thorstenbeier/src/pyjs/emsdk_install/upstream/emscripten/cache/sysroot/include/c++/v1/__memory/pointer_traits.h:195:59: note: during template argument deduction for class template partial specialization '_HasToAddress<_Pointer, decltype((void)pointer_traits<_Pointer>::to_address(std::declval<const _Pointer &>()))>' [with _Pointer = sparrow::test_iterator]
  195 |   static const bool value = _HasArrow<_Pointer>::value || _HasToAddress<_Pointer>::value;
      |                                                           ^
/Users/thorstenbeier/src/pyjs/emsdk_install/upstream/emscripten/cache/sysroot/include/c++/v1/__memory/pointer_traits.h:195:59: note: in instantiation of template class 'std::_HasToAddress<sparrow::test_iterator>' requested here
/Users/thorstenbeier/src/pyjs/emsdk_install/upstream/emscripten/cache/sysroot/include/c++/v1/__type_traits/conjunction.h:27:32: note: in instantiation of template class 'std::_IsFancyPointer<sparrow::test_iterator>' requested here
   27 | __expand_to_true<__enable_if_t<_Pred::value>...> __and_helper(int);
      |                                ^
/Users/thorstenbeier/src/pyjs/emsdk_install/upstream/emscripten/cache/sysroot/include/c++/v1/__type_traits/conjunction.h:38:39: note: while substituting explicitly-specified template arguments into function template '__and_helper'
   38 | using _And _LIBCPP_NODEBUG = decltype(std::__and_helper<_Pred...>(0));
      |                                       ^
/Users/thorstenbeier/src/pyjs/emsdk_install/upstream/emscripten/cache/sysroot/include/c++/v1/__memory/pointer_traits.h:200:5: note: (skipping 2 contexts in backtrace; use -ftemplate-backtrace-limit=0 to see all)
  200 |     _And<is_class<_Pointer>, _IsFancyPointer<_Pointer> >::value
      |     ^
/Users/thorstenbeier/src/pyjs/emsdk_install/upstream/emscripten/cache/sysroot/include/c++/v1/__memory/pointer_traits.h:235:59: note: while substituting deduced template arguments into function template '__to_address' [with _Pointer = sparrow::test_iterator, $1 = (no value)]
  235 | auto to_address(const _Pointer& __p) noexcept -> decltype(std::__to_address(__p)) {
      |                                                           ^
/Users/thorstenbeier/src/pyjs/emsdk_install/upstream/emscripten/cache/sysroot/include/c++/v1/__iterator/concepts.h:193:7: note: while substituting deduced template arguments into function template 'to_address' [with _Pointer = sparrow::test_iterator]
  193 |     { _VSTD::to_address(__i) } -> same_as<add_pointer_t<iter_reference_t<_Ip>>>;
      |       ^
/Users/thorstenbeier/src/pyjs/emsdk_install/upstream/emscripten/cache/sysroot/include/c++/v1/__config:666:17: note: expanded from macro '_VSTD'
  666 | #  define _VSTD std
      |                 ^
/Users/thorstenbeier/src/pyjs/emsdk_install/upstream/emscripten/cache/sysroot/include/c++/v1/__iterator/concepts.h:193:7: note: in instantiation of requirement here
  193 |     { _VSTD::to_address(__i) } -> same_as<add_pointer_t<iter_reference_t<_Ip>>>;
      |       ^~~~~~~~~~~~~~~~~~~~~~
/Users/thorstenbeier/src/pyjs/emsdk_install/upstream/emscripten/cache/sysroot/include/c++/v1/__config:666:17: note: expanded from macro '_VSTD'
  666 | #  define _VSTD std
      |                 ^
/Users/thorstenbeier/src/pyjs/emsdk_install/upstream/emscripten/cache/sysroot/include/c++/v1/__iterator/concepts.h:192:3: note: while substituting template arguments into constraint expression here
  192 |   requires(const _Ip& __i) {
      |   ^~~~~~~~~~~~~~~~~~~~~~~~~~
  193 |     { _VSTD::to_address(__i) } -> same_as<add_pointer_t<iter_reference_t<_Ip>>>;
      |     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  194 |   };
      |   ~
/Users/thorstenbeier/src/sparrow/test/test_iterator.cpp:241:36: note: while checking the satisfaction of concept 'contiguous_iterator<sparrow::test_iterator>' requested here
  241 |             constexpr bool valid = std::contiguous_iterator<test_iterator>;
      |                                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/thorstenbeier/src/pyjs/emsdk_install/upstream/emscripten/cache/sysroot/include/c++/v1/__memory/pointer_traits.h:38:8: note: template is declared here
   38 | struct __pointer_traits_element_type;
```

Reply:
It's the same error as the one we worked around with std::equal.

Reply:
Jeah, but strange that this is only triggered in emscripten

Reply:
Ok so the `#include <version>` fix could help too right?